bugsnag/middleware.py

```diff
@@ -104,6 +104,38 @@ def append(self, middleware):
         """
         self.stack.append(middleware)
 
+    def insert_before(self, middleware, target_class):
```

Comment:
It might be more conventional to swap the order of the parameters so that the target comes before the middleware to add. This closer matches how bugsnag-ruby's middleware stack, Rails middleware stack, and Python's `list.insert` method work.